test/Behodler.test.js

```diff
@@ -1,102 +1,153 @@
+        this.pyroRegular.deployed()
+
+        this.dai = await MockToken1.deploy()
+        this.dai.deployed()
+
+        this.burnableToken = await MockToken1.deploy()
+        this.burnableToken.deployed()
+
+        this.eye = await MockToken1.deploy()
+        this.eye.deployed()
+
+        this.invalidToken = await MockToken1.deploy()
+        this.invalidToken.deployed()
+
+        const OpenArbiter = await ethers.getContractFactory('OpenArbiter')
+        this.flashLoanArbiter = await OpenArbiter.deploy()
+        this.flashLoanArbiter.deployed()
+
+        await this.regularToken.mint(trader1.address, 2000000n * TEN)
+        await this.burnableToken.mint(trader1.address, 2000000n * TEN)
+        await this.invalidToken.mint(trader1.address, TEN)
+
+        const MockWeiDai = await ethers.getContractFactory("MockWeiDai")
+        this.mockWeiDai = await MockWeiDai.deploy()
+        this.mockWeiDai.deployed()
+
+        await this.behodler.seed(this.weth.address, this.lachesis.address, this.flashLoanArbiter.address, this.liquidityReceiver.address, weiDaiReserve.address, this.dai.address, this.mockWeiDai.address)
+        await this.behodler.configureScarcity(110, 25, feeDestination.address)
+        await this.lachesis.measure(this.regularToken.address, true, false)
+        await this.lachesis.measure(this.burnableToken.address, true, true)
+
+        await this.lachesis.setBehodler(this.behodler.address)
+        await this.lachesis.updateBehodler(this.regularToken.address)
+        await this.lachesis.updateBehodler(this.burnableToken.address)
     })
 
     it('adding burnable token as liquidity in 2 batches generates the correct volume of Scarcity', async function () {
         //ADD 1 FINNEY WHEN BEHODLER BALANCE OF TOKEN ZERO
         const originalBalance = 2000000n * TEN
         const expectedBalanceAfter = originalBalance - FINNEY * 2n
 
-        const scarcitySupplyBefore = await bigNum.BNtoBigInt(this.behodler.totalSupply.call())
+        const scarcitySupplyBefore = await bigNum.BNtoBigInt(this.behodler.totalSupply())
         expect(scarcitySupplyBefore).to.equal(0n)
 
-        await this.burnableToken.approve(this.behodler.address, originalBalance, { from: trader1 })
-        await this.behodler.addLiquidity(this.burnableToken.address, FINNEY * 2n, { from: trader1 })
-        const tokenBalanceOfUser = await bigNum.BNtoBigInt(this.burnableToken.balanceOf.call(trader1))
-        expect(tokenBalanceOfUser).to.equal(expectedBalanceAfter)
+        await this.burnableToken.connect(trader1).approve(this.behodler.address, originalBalance)
+        await this.behodler.connect(trader1).addLiquidity(this.burnableToken.address, FINNEY * 2n)
 
-        const scarcityBalance = (await this.behodler.balanceOf.call(trader1)).toString()
+        const tokenBalanceOfUser = await bigNum.BNtoBigInt(this.burnableToken.balanceOf(trader1.address))
```

Comment:
await bigNum.BNtoBigInt(this.burnableToken.balanceOf(trader1.address)) -->
await this.burnableToken.balanceOf(trader1.address) and other places.
You are doing all needed manipulations with BN inside assertBNequal

Reply:
changed where it possible not to fail the tests